homeassistant/components/playstation_network/media_player.py

```diff
@@ -0,0 +1,145 @@
+"""Media player entity for the PlayStation Network Integration."""
+
+from enum import StrEnum
+import logging
+
+from homeassistant.components.media_player import (
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaPlayerState,
+    MediaType,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import PlaystationNetworkConfigEntry, PlaystationNetworkCoordinator
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class PlatformType(StrEnum):
+    """PlayStation Platform Enum."""
+
+    PS5 = "PS5"
+    PS4 = "PS4"
+
+
+PLATFORM_MAP = {
+    PlatformType.PS5: "PlayStation 5",
+    PlatformType.PS4: "PlayStation 4",
+}
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: PlaystationNetworkConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Media Player Entity Setup."""
+    coordinator = config_entry.runtime_data
+    devices_added: set[PlatformType] = set()
+    supported_devices = {PlatformType.PS4, PlatformType.PS5}
+    device_reg = dr.async_get(hass)
+    entities = []
+
+    @callback
+    def add_entities() -> None:
+        nonlocal devices_added
+
+        if not supported_devices - devices_added:
+            remove_listener()
+
+        if (platform := coordinator.data.platform["platform"]) and (
+            platform_type := PlatformType(platform)
+        ) not in devices_added:
+            async_add_entities([MediaPlayer(coordinator, platform_type)])
+            devices_added.add(platform_type)
+
+    for platform in supported_devices:
+        if device_reg.async_get_device(
+            identifiers={(DOMAIN, f"{coordinator.config_entry.unique_id}_{platform}")}
+        ):
+            entities.append(MediaPlayer(coordinator, platform))
+            devices_added.add(platform)
+    if entities:
+        async_add_entities(entities)
+
+    remove_listener = coordinator.async_add_listener(add_entities)
+    add_entities()
+
+
+class MediaPlayer(CoordinatorEntity[PlaystationNetworkCoordinator], MediaPlayerEntity):
```

Comment:
I think the `MediaPlayer` class name might be confusing. Maybe `PsnMediaPlayerEntity`?